src/script/view_model/ContentViewModel.ts

```diff
@@ -133,10 +134,112 @@ export class ContentViewModel {
     }
   }
 
-  changeConversation = (conversationEntity: Conversation, messageEntity?: Message) => {
+  private changeConversation(conversationEntity: Conversation, messageEntity?: Message): void {
     this.initialMessage = messageEntity;
     this.conversationState.activeConversation(conversationEntity);
-  };
+  }
+
+  private async getConversationEntity(
+    conversation: Conversation | string,
+    domain: string | null = null,
+  ): Promise<Conversation> {
+    if (isConversationEntity(conversation)) {
+      return conversation;
+    }
+    return await this.conversationRepository.getConversationById({domain: domain || '', id: conversation});
+  }
+
+  private closeRightSidebar(): void {
+    const {rightSidebar} = useAppMainState.getState();
+    rightSidebar.close();
+  }
+
+  private handleMissingConversation(): void {
+    this.closeRightSidebar();
+    return this.switchContent(ContentState.CONNECTION_REQUESTS);
+  }
+
+  private closeRightSidebarAndThrowError(): void {
+    this.closeRightSidebar();
+    throw new ConversationError(
+      ConversationError.TYPE.CONVERSATION_NOT_FOUND,
+      ConversationError.MESSAGE.CONVERSATION_NOT_FOUND,
+    );
+  }
+
+  private isConversationOpen(conversationEntity: Conversation, isActiveConversation: boolean): boolean {
+    const {contentState} = useAppState.getState();
+    const isConversationState = contentState === ContentState.CONVERSATION;
+    return conversationEntity && isActiveConversation && isConversationState;
+  }
+
+  private switchToNotificationSettingsIfApplicable(
+    openNotificationSettings: boolean,
+    conversationEntity: Conversation,
+  ): void {
+    if (openNotificationSettings) {
+      const {rightSidebar} = useAppMainState.getState();
+      rightSidebar.goTo(PanelState.NOTIFICATIONS, {entity: conversationEntity});
+    }
+  }
+
+  private handleConversationState(
+    isOpenedConversation: boolean,
+    openNotificationSettings: boolean,
+    conversationEntity: Conversation,
+  ): boolean {
+    const {setContentState} = useAppState.getState();
+    if (isOpenedConversation) {
+      this.switchToNotificationSettingsIfApplicable(openNotificationSettings, conversationEntity);
+      return true;
+    }
+    setContentState(ContentState.CONVERSATION);
+
+    this.mainViewModel.list.openConversations(conversationEntity.archivedState());
+
+    return false;
+  }
+
+  private processMessageEntity(
+    conversationEntity: Conversation,
+    openFirstSelfMention: boolean,
+    exposeMessageEntity: Message | undefined,
+  ): void {
+    const messageEntity = openFirstSelfMention ? conversationEntity.getFirstUnreadSelfMention() : exposeMessageEntity;
+    if (conversationEntity.is_cleared()) {
+      conversationEntity.cleared_timestamp(0);
+    }
+    this.changeConversation(conversationEntity, messageEntity);
+  }
```

Comment:
Doesn't really seem to be processing message, is it? I don't think I understand the name of the function 🤔

Reply:
It's determining the message entity to work with based on the 2 criteria. So I named it processMessageEntity. But you are right its not processing any message so a better name would be updateConversationStatusBasedOnMessage or handleConversationWithMessageEntity. what do you think?

Reply:
is it finding the message to focus when opening the conversation? I feel this is what it's doing. The name should reflect that I guess. 

Aslo it's not clear to me why this particular function is changing the `cleared_timestamp` I feel it's not its responsibility (should probably be done in the glue function `showConversation`)

Reply:
i have removed this method, cleared_timestamp is not a responsibility of this function so put it back in the glue function. Now  this.changeConversation(conversationEntity, messageEntity); is already a method so putting it back to the glue function too.